include/aikido/planner/World.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef AIKIDO_PLANNER_WORLD_HPP_
+#define AIKIDO_PLANNER_WORLD_HPP_
+
+#include <string>
+#include <dart/dart.hpp>
+
+namespace aikido {
+namespace planner {
+
+class World
+{
+public:
+  /// Construct a kinematic World.
+  World(const std::string& name = "");
+
+  virtual ~World();
+
+  /// Create a clone of this World. All Skeletons will be copied over.
+  /// \param newName Name for the cloned World
+  std::shared_ptr<World> clone(const std::string& newName = "") const;
+
+  /// Create a new World inside of a shared_ptr
+  static std::shared_ptr<World> create(const std::string& name = "");
+
+  /// Set the name of this World
+  std::string setName(const std::string& newName);
+
+  /// Get the name of this World
+  const std::string& getName() const;
+
+  /// Find a Skeleton by index
+  dart::dynamics::SkeletonPtr getSkeleton(std::size_t i) const;
+
+  /// Find a Skeleton by name
+  dart::dynamics::SkeletonPtr getSkeleton(const std::string& name) const;
+
+  /// Get the number of Skeletons
+  std::size_t getNumSkeletons() const;
+
+  /// Add a Skeleton to this World
+  std::string addSkeleton(const dart::dynamics::SkeletonPtr& skeleton);
+
+  /// Remove a Skeleton from this World
+  void removeSkeleton(const dart::dynamics::SkeletonPtr& skeleton);
+
+  // TODO: Add methods for registering callbacks?
+
+  // Get the mutex that protects the state of this World.
+  std::mutex& getMutex() const;
+
+protected:
+  /// Name of this World
+  std::string mName;
+
+  /// Skeletons in this World
+  std::vector<dart::dynamics::SkeletonPtr> mSkeletons;
+
+  mutable std::mutex mMutex;
+
+  /// NameManager for keeping track of Worlds
+  static dart::common::NameManager<World*> worldNameManager;
+
+  /// NameManager for keeping track of Skeletons
+  dart::common::NameManager<dart::dynamics::SkeletonPtr> mSkeletonNameManager;
+};
+
+typedef std::shared_ptr<World> WorldPtr;
```

Comment:
Nit: Let's use C++11 style of type aliasing: `using WorldPtr = std::shared_ptr<World>()`.